src/API/Management/ClientGrants.php

```diff
@@ -2,75 +2,193 @@
 
 namespace Auth0\SDK\API\Management;
 
-use Auth0\SDK\API\Header\ContentType;
+use Auth0\SDK\Exception\CoreException;
 
+/**
+ * Class ClientGrants.
+ * Handles requests to the Client Grants endpoint of the v2 Management API.
+ *
+ * @package Auth0\SDK\API\Management
+ */
 class ClientGrants extends GenericResource
 {
+
     /**
+     * Get all Client Grants, by page if desired.
+     * Required scope: "read:client_grants"
+     *
+     * @param array        $params   Additional URL parameters to send:
+     *      - "audience" to filter be a specific API audience identifier.
+     *      - "client_id" to return an object.
+     *      - "include_totals" to return an object.
+     * @param null|integer $page     The page number, zero based.
+     * @param null|integer $per_page The amount of entries per page.
      *
-     * @param  string      $id
-     * @param  null|string $audience
      * @return mixed
+     *
+     * @throws \Exception Thrown by the Guzzle HTTP client when there is a problem with the API call.
+     *
+     * @link https://auth0.com/docs/api/management/v2#!/Client_Grants/get_client_grants
      */
-    public function get($id, $audience = null)
+    public function getAll(array $params = [], $page = null, $per_page = null)
     {
-        $request = $this->apiClient->get()
-        ->addPath('client-grants');
+        if (null !== $page) {
+            $params['page'] = abs(intval($page));
+        }
 
-        if ($audience !== null) {
-            $request = $request->withParam('audience', $audience);
+        if (null !== $per_page) {
+            $params['per_page'] = abs(intval($per_page));
         }
 
-        return $request->call();
+        return $this->apiClient->method('get')
+            ->addPath('client-grants')
+            ->withDictParams($params)
+            ->call();
+    }
+
+    /**
+     * Get Client Grants by audience.
+     * Required scope: "read:client_grants"
+     *
+     * @param string       $audience API Audience to filter by.
+     * @param null|integer $page     The page number, zero based.
+     * @param null|integer $per_page The amount of entries per page.
+     *
+     * @return mixed
+     *
+     * @throws CoreException Thrown when $audience is empty or not a string.
+     * @throws \Exception Thrown by the Guzzle HTTP client when there is a problem with the API call.
+     *
+     * @link https://auth0.com/docs/api/management/v2#!/Client_Grants/get_client_grants
+     */
+    public function getByAudience($audience, $page = null, $per_page = null)
+    {
+        if (empty($audience) || ! is_string($audience)) {
```

Comment:
that exclamation mark

Reply:
So the rest of the code follows this, although I think it's ugly. If it's consistent with the rest of the code base, then I'm fine with it.

Reply:
"Ugly" as in you don't like the error checking here? @lbalmaceda and I talked about failing early in methods to avoid API errors. For this one, an empty `$audience` would not fail but it also would not filter so the method would be useless.

Reply:
Don't like the space between a `!` and the context. But don't have any strong feelings as all the others are like this in the PR so presume it's inline with coding style of the SDK.

Reply:
I was on the fence about the space. This is all whitespace-corrected now so I can enforce no space.